feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/repository/RealProxyRepository.kt

```diff
@@ -0,0 +1,92 @@
+package io.novafoundation.nova.feature_account_impl.data.repository
+
+import io.novafoundation.nova.common.address.AccountIdKey
+import io.novafoundation.nova.common.address.intoKey
+import io.novafoundation.nova.common.data.network.runtime.binding.castToDictEnum
+import io.novafoundation.nova.common.data.network.runtime.binding.castToList
+import io.novafoundation.nova.common.data.network.runtime.binding.castToStruct
+import io.novafoundation.nova.common.data.network.runtime.binding.getTyped
+import io.novafoundation.nova.common.utils.Modules
+import io.novafoundation.nova.feature_account_api.data.model.ProxiedWithProxies
+import io.novafoundation.nova.feature_account_api.data.repository.ProxyRepository
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccountId
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import io.novafoundation.nova.runtime.storage.source.StorageDataSource
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module
+import jp.co.soramitsu.fearless_utils.runtime.metadata.storage
+
+class RealProxyRepository(
+    private val remoteSource: StorageDataSource
+) : ProxyRepository {
+
+    override suspend fun getProxyDelegatorsForAccounts(chainId: ChainId, metaAccountIds: List<MetaAccountId>): List<ProxiedWithProxies> {
+        val delegatorToProxies = receiveAllProxies(chainId)
+
+        val accountIdToMetaAccounts = metaAccountIds.associateBy { it.accountId.intoKey() }
```

Comment:
Does it mean there will only be a single matched MetaAccount? I think its better to use `groupBy`

Reply:
As I understand we need this to support case when we added the same account several times, right?

Reply:
Yes, I remember we agreed that there should be a separate wallet for each such case